cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ValidationHandler.java

```diff
@@ -141,32 +119,31 @@ public void validateStage(HttpServiceRequest request, HttpServiceResponder respo
       }
     }
 
-    // evaluate secure macros
+    Function<Map<String, String>, Map<String, String>> macroFn = getMacroEvaluationFunction(namespace, arguments);
+    String validationResponse = ValidationUtils
+      .validate(validationRequest, getContext().createPluginConfigurer(namespace), macroFn);
+    responder.sendString(validationResponse);
+  }
+
+  private Function<Map<String, String>, Map<String, String>> getMacroEvaluationFunction(
```

Comment:
Please consider extracting a lambda one level up. It's usually easier to follow without functions that return functions.
E.g. make `evaluateMacros(namespace, programArgs, macroProperties)` and then  
`macroFn = macroProperties -> evaluateMacros(namespace, arguments, macroProperties)`

Reply:
Done.